cmake/common/px4_base.cmake

```diff
@@ -701,14 +701,11 @@ function(px4_add_common_flags)
 		-DCONFIG_ARCH_BOARD_${board_config}
 		)
 
-	if (NOT ${CMAKE_C_COMPILER_ID} MATCHES ".*Clang.*")
-		set(added_exe_link_flags
+	if (NOT (UNIX AND APPLE))
```

Comment:
This is detection has changed from a compiler check to a OS check.

Refering to http://www.openguru.com/2009/04/cmake-detecting-platformoperating.html

If this does not work on Clang on, OSX,  only on travis the that shouldn't that be what is detected?